sstables/mx/reader.cc

```diff
@@ -1737,6 +1737,13 @@ mutation_reader make_reader(
             value_or_reference(std::move(slice)), std::move(trace_state), fwd, fwd_mr, monitor);
 }
 
+/// a reader which does not support seeking to given position.
+///
+/// unlike max_sstable_mutation_reader which allows positioned read,
+/// mx_crawling_sstable_mutation_reader only supports sequential read. It is designed
```

Comment:
All readers only support sequential read. The difference between the crawling vs. the normal reader is:
* crawling reader always reads full range -> not possible to read a subset of the file
* crawling reader doesn't support fast forwarding (either kind)

Reply:
sorry for the confusion. i am using the terminology of  sequential read versus seek (considering the disk i/o system in general)

Reply:
I think using general IO terminology is confusing, because we don't support the usual seek/tell operations. Seek works in any direction, while in sstable readers, you can only move forward, and only to a range which doesn't intersect with the current read range. Also, you have to provide the read range up-front (including end). If you don't provide an end bound, you read to the end of the file and fast-forward is not supported. So, as you see there are a lot of constraints here.

Reply:
i see. indeed, we don't rewind. will incorporate your comment at https://github.com/scylladb/scylladb/pull/20185#discussion_r1721327060 in the next revision in #20187